contracts/finality/src/exec/finality.rs

```diff
@@ -30,6 +30,11 @@ pub fn handle_public_randomness_commit(
     commitment: &[u8],
     signature: &[u8],
 ) -> Result<Response<BabylonMsg>, ContractError> {
+    // Validate num_pub_rand is at least 1 to prevent integer underflow
+    if num_pub_rand == 0 {
```

Comment:
Should there be a maximum?

Reply:
The maximum does not seem to matter. For honest FPs they won't commit excessive pub rand at one go. For adversarial FPs doing this will burn more gas, yet does not break the protocol's security. CosmWasm / Cosmos SDK also has gas limit so finality sig with super big Merkle proof won't go through